spec/models/validations/sales/setup_validations_spec.rb

```diff
@@ -6,43 +6,95 @@
   let(:validator_class) { Class.new { include Validations::Sales::SetupValidations } }
 
   describe "#validate_saledate" do
-    context "when saledate is blank" do
-      let(:record) { FactoryBot.build(:sales_log, saledate: nil) }
+    context "with sales_in_crossover_period == true" do
+      before do
+        allow(FormHandler.instance).to receive(:sales_in_crossover_period?).and_return(true)
+      end
```

Comment:
Maybe mock the date to be in crossover period instead, because I think the previous, current and next forms will change when we enter the crossover period